### ISSUE TITLE
Avoid accessing GitHub directly

### ISSUE THREAD

**Issue by trieloff** [adobe/helix-static#348]
In the light of today's outage I thought how we could put another cache between helix-static and raw.

We have almost all the pieces in place to use Fastly as a GitHub proxy and don't even need to change the VCL. All that's needed is for helix-static to make requests to the original host (Fastly) and pass in a X-Request-Type and X-Location header. The request type should be the same we use for redirects in case of 1 MB static resources. The X-Location is the current raw.githubusercontent.com URL. 

Fastly and Runtime should cache the responses (which are immutable) for a long time, giving us some robustness for the next GitHub outage.

**Comment by dominique-pfister** [comment]
Interesting 😄  Do I interpret the flow correctly: a request goes through Fastly, then eventually hits `helix-static` which again goes back to Fastly? If so, how do we ensure that no bogus requests containing those two headers will be considered genuinely originating from `helix-static` by Fastly?

**Comment by trieloff** [comment]
Yes, it would be similar to the flow we have for dynamic content where it's Fastly -> Runtime (Pipeline) -> Fastly, again -> Runtime (Content Proxy) -> GitHub – we ensure that no rendering-critical component is ever talking to GitHub directly.

https://github.com/adobe/helix-publish/blob/926d1a5e1d0427ddb60fe6e225e6ed1165ad7641/layouts/fastly/helix.vcl#L928-L929

The two headers we need are: `X-Request-Type: Static/Redirect` and `X-Backend-URL: /owner/repo/ref/path`. You can already use them and your site will just deliver the same content as `raw.githubusercontent.com` would. I don't think there is much danger in people using this on arbitrary sites, and when it becomes an issue, we can validate the client IP range.

**Comment by tripodsan** [comment]
why not having helix-dispatch call static via fastly?

**Comment by trieloff** [comment]
This would require bigger changes:
- helix-dispatch is not making HTTP requests right now, just using ow.action.invoke with different configurations 
- we would effectively roll helix-static into dispatch or we'd have to change the VCL so that there is a direct request type that goes to helix-static

**Comment by trieloff** [comment]
It works … almost:

https://github.com/adobe/helix-publish/blob/311be5854831d9791ea20a4e382aaaf84bd167e6/layouts/fastly/helix.vcl#L116-L129

will reset the two headers we need, so I suggest, we

1. relax the condition in `helix-publish`
2. bump the `helix-static` and `helix-dispatch` major versions (as they will require a matching VCL)
3. adjust the VCL to use the new major version
